### ISSUE TITLE
Failure "Ssl_method SSLv23_method not implemented"

### ISSUE THREAD

**Issue by alinpopa** [janestreet/async_ssl#27]
Getting the following exception when using the `v0.9` branch, on Arch linux:
```
(monitor.ml.Error (Failure "Ssl_method SSLv23_method not implemented")
  ("Raised at file \"pervasives.ml\", line 32, characters 17-33"
    "Called from file \"src/ffi.ml\", line 117, characters 23-44"
    "Called from file \"src/ssl.ml\", line 425, characters 14-44"
    "Called from file \"src/memo.ml\", line 13, characters 13-18"
    "Re-raised at file \"src/memo.ml\" (inlined), line 10, characters 16-23"
    "Called from file \"src/memo.ml\", line 34, characters 5-121"
    "Called from file \"src/ssl.ml\", line 443, characters 4-58"
    "Called from file \"src/monitor.ml\", line 271, characters 42-51"
    "Called from file \"src/job_queue.ml\", line 159, characters 6-47"))
```
Could be any prerequisite that I've forgot?

Thanks.

**Comment by trefis** [comment]
When you say "the `v0.9` branch", you mean the branch from this repository right? Not the `v0.9` opam package?

**Comment by alinpopa** [comment]
Yes, that's the one I'm talking about; cloned this repo, switched to that branch, and pinned it using `opam pin add ... `.

**Comment by brendanlong** [comment]
It looks like the newest version of OpenSSL renames `SSLv23_method` to the more-accurate `TLS_method`, then aliases it with a macro (which `async_ssl` can't detect). I'm working on a patch to make it look for both.

**Comment by dlobraico** [comment]
I think this was resolved in #28.
